### ISSUE TITLE
Joining relationships twice causes SQL Exception.

### ISSUE THREAD

**Issue by timrosskamp** [kirschbaum-development/eloquent-power-joins#37]
This code produces an SQL Exception, because the relationship is joined twice in the query:

```php
Model::query()
  ->joinRelationship('relation')
  ->joinRelationship('relation')
  ->dd();
```

outputs:

```sql
select `model`.* from `model` inner join `relation` on `relation`.`id` = `model`.`id` inner join `relation` on `relation`.`id` = `model`.`id`
```

I've tried to get to the bottom of the bug and for me it seems like the `PowerJoins::$joinRelationshipCache` cache gets emptied for some reason.

But this only happens in this case. When joining two nested relationships (as in the test case) the sql query is correct, that's why this bug wasn't covered by the tests.

**Comment by timrosskamp** [comment]
https://github.com/kirschbaum-development/eloquent-power-joins/commit/7c1ce762f103ea4d86e25effc92e4628c95950e2 must be related to this issue. Before this commit, I didn't have this problem.

**Comment by luisdalmolin** [comment]
@timrosskamp Ok, this seems a valid issue indeed, I thought we had test coverage and this would be a non-issue after that commit. I'll take a look at this week.

For now, if you lock the version as `2.2.2` you should be good.

**Comment by luisdalmolin** [comment]
@timrosskamp This issue should be fixed in version 2.3.0.
